### ISSUE TITLE
Optimize the prioritytarget view

### ISSUE THREAD

**Issue by rachel3834** [LCOGT/mop#130]
Feedback from Jashandeep:

Digging into the /prioritytargets/ view, it does look like there's some significant processing (time-wise) that takes place after the query: https://github.com/LCOGT/mop/blob/aca606d89a22ec84bb307299e9b01746e3cec719/mop/views.py#L192
# Repackage the two lists to extract the parameters to display in the table.

There's probably some optimizations here that could speed things up. For instance doing DB look-up in a for loop is going to be slow: https://github.com/LCOGT/mop/blob/aca606d89a22ec84bb307299e9b01746e3cec719/mop/views.py#L216
target = Target.objects.filter(pk=target_id[0])[0]

**Comment by rachel3834** [comment]
For the /targets/<id>/ page, these lines should be moved to the conditional, if possible: https://github.com/LCOGT/mop/blob/aca606d89a22ec84bb307299e9b01746e3cec719/mop/views.py#L37-L38
[views.py](https://github.com/LCOGT/mop/blob/aca606d89a22ec84bb307299e9b01746e3cec719/mop/views.py)
        target = self.get_object()
        set_target_sky_location(target)
<https://github.com/[LCOGT/mop](https://github.com/LCOGT/mop)|LCOGT/mop>LCOGT/mop | Added by [GitHub](https://lcogt.slack.com/services/B01UJTCARGE)
[9:39](https://lcogt.slack.com/archives/CSN3T78QZ/p1707500391114989)
That function does a DB save every call: https://github.com/LCOGT/mop/blob/aca606d89a22ec84bb307299e9b01746e3cec719/mop/toolbox/TAP.py#L108
[TAP.py](https://github.com/LCOGT/mop/blob/aca606d89a22ec84bb307299e9b01746e3cec719/mop/toolbox/TAP.py)
def set_target_sky_location(target):
<https://github.com/[LCOGT/mop](https://github.com/LCOGT/mop)|LCOGT/mop>LCOGT/mop | Added by [GitHub](https://lcogt.slack.com/services/B01UJTCARGE)
[9:40](https://lcogt.slack.com/archives/CSN3T78QZ/p1707500405458769)
Which can't be good for performance

**Comment by rachel3834** [comment]
I've implemented the proposed optimizations as described above so that we are not doing as many DB queries in loops.

**Comment by rachel3834** [comment]
Timing data suggests this may roughly half the load time for the page.

**Comment by rachel3834** [comment]
Checkpointing the existing code before further optimizations:

```
PRIORITYTARGETS started context get 2024-02-22 22:07:33.620458
CHECKPOINT: N DB connections: 1, memory: 248.02MiB
/data/software/mop_venv/lib/python3.10/site-packages/django/views/generic/list.py:91: UnorderedObjectListWarning:

Pagination may yield inconsistent results with an unordered object_list: <class 'tom_targets.models.TargetExtra'> QuerySet.

Priority querysets initially find 2 stellar candidate events and 19 BH candidate events
PRIORITYTARGETS started extract at 2024-02-22 22:07:33.734215
CHECKPOINT: N DB connections: 11, memory: 249.38MiB
PRIORITYTARGETS finished data extract at 2024-02-22 22:07:34.097975
PRIORITY TARGETS extract function took 0:00:00.363760
CHECKPOINT: N DB connections: 105, memory: 250.5MiB
PRIORITYTARGETS started extract at 2024-02-22 22:07:34.098766
CHECKPOINT: N DB connections: 105, memory: 250.5MiB
PRIORITYTARGETS finished data extract at 2024-02-22 22:07:36.690095
PRIORITY TARGETS extract function took 0:00:02.591329
CHECKPOINT: N DB connections: 697, memory: 255.98MiB
After filtering priority targets, returning 2 stellar candidate events and 12 BH candidate events
PRIORITYTARGETS finished context get 2024-02-22 22:07:36.691680 took 0:00:03.071222
CHECKPOINT: N DB connections: 697, memory: 255.98MiB
FINISHED GET_CONTEXT 2024-02-22 22:07:37.105672
CHECKPOINT: N DB connections: 697, memory: 270.97MiB
```

**Comment by rachel3834** [comment]
Revised code is more efficient on querysets and a bit quicker.  Major changes include properly searching for targets that are classified as microlensing, alive, not QSOs etc instead of doing this filtering in a later for loop.  
I've added checkpoints so I can gauge how this behaves in operation.

**Comment by rachel3834** [comment]
A significant issue with this refactor is that all non-Gaia targets have disappeared from the board.  This is addressed in Issue #141
